src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java

```diff
@@ -221,6 +275,19 @@ private String getStepName(@Nonnull StepAtomNode node, @Nonnull String name) {
             return name;
         }
         UninstantiatedDescribable describable = getUninstantiatedDescribableOrNull(node, stepDescriptor);
+        return getDisplayName(stepDescriptor, describable);
+    }
+
+    private String getStepName(@Nonnull StepStartNode node, @Nonnull String name) {
+        StepDescriptor stepDescriptor = node.getDescriptor();
+        if (stepDescriptor == null) {
```

Comment:
Should we support also blank strings?
```suggestion
        if (StringUilts.isBlank(stepDescriptor)) {
```

Reply:
why?